tests/chainer_tests/test_function.py

```diff
@@ -316,4 +316,50 @@ def test_forward_invalid(self):
             f(v)
 
 
+class TestFunctionDebug(unittest.TestCase):
+
+    def setUp(self):
+        chainer.set_debug(True)
+        self.one = numpy.array([1], numpy.float32)
+        self.nan = numpy.array([float('nan')], numpy.float32)
+        self.f = chainer.Function()
+
+    def tearDown(self):
+        chainer.set_debug(False)
```

Comment:
Original debug mode not necessarily `False`. So please store current debug mode in `setUp` and restore it in `tearDown`.